pkg/restmapper/dynamicrestmapper.go

```diff
@@ -54,14 +88,19 @@ func (drm *DynamicRESTMapper) reload() error {
 	return nil
 }
 
-// reloadOnError checks if an error indicates that the delegated RESTMapper needs to be
-// reloaded, and if so, reloads it and returns true.
+// reloadOnError checks if an error indicates that the delegated RESTMapper
+// needs to be reloaded, and if so, reloads it and returns true.
+// reloadOnError uses an exponential backoff mechanism to rate limit reloads.
 func (drm *DynamicRESTMapper) reloadOnError(err error) bool {
 	if _, matches := err.(*meta.NoKindMatchError); !matches {
+		drm.backoff.reset()
 		return false
 	}
 	err = drm.reload()
 	if err != nil {
+		// TODO(estroz): HandleError uses a rudimentary backoff by default.
+		// Should we remove it completely or substitute the default backoff
```

Comment:
I think it makes sense to have the simple back off as a fall-back before failing completely